### PR TITLE
buildSandbox: return the exit status of wrapped executable

### PR DESCRIPTION
**Pull request by Profpatsch**
Since we want to create a transparent sandbox, returning the same exit
code as the wrapped process is very important.

### CODE REVIEW

File: pkgs/build-support/build-sandbox/src/setup.c [hunk 1]
```diff
@@ -864,9 +864,27 @@ bool setup_sandbox(void)
...
      * otherwise we can't mount /proc in the right PID namespace.
      */
+    int wstatus;
+    int w;
     if (pid > 0) {
-        waitpid(pid, NULL, 0);
-        _exit(1);
+
+        w = waitpid(pid, &wstatus, 0);
+        if (w == -1) {
```
Comment (aszlig) [inline pkgs/build-support/build-sandbox/src/setup.c]:
```suggestion
        if (waitpid(pid, &wstatus, 0) == -1) {
```

File: pkgs/build-support/build-sandbox/src/setup.c [hunk 2]
```diff
@@ -864,9 +864,27 @@ bool setup_sandbox(void)
...
+        else if (WIFEXITED(wstatus)) {
+          _exit(WEXITSTATUS(wstatus));
+        }
+        else if (WIFSIGNALED(wstatus)) {
+          fprintf(stderr, "sandbox: killed by signal %d", WTERMSIG(wstatus));
+          _exit(EXIT_FAILURE);
+        }
+        else {
+          // WIFSTOPPED, WIFCONTINUED?
+          fprintf(stderr, "sandbox: wait failed");
```
Comment (aszlig) [inline pkgs/build-support/build-sandbox/src/setup.c]:
```suggestion
          fputs("sandbox: wait failed\n", stderr);
```

File: pkgs/build-support/build-sandbox/src/setup.c [hunk 3]
```diff
@@ -864,9 +864,27 @@ bool setup_sandbox(void)
...
+        w = waitpid(pid, &wstatus, 0);
+        if (w == -1) {
+          fprintf(stderr, "sandbox: waitpid failure");
+          _exit(EXIT_FAILURE);
+        }
+        else if (WIFEXITED(wstatus)) {
+          _exit(WEXITSTATUS(wstatus));
+        }
+        else if (WIFSIGNALED(wstatus)) {
+          fprintf(stderr, "sandbox: killed by signal %d", WTERMSIG(wstatus));
```
Comment (aszlig) [inline pkgs/build-support/build-sandbox/src/setup.c]:
```suggestion
          fprintf(stderr, "sandbox: killed by signal %d\n", WTERMSIG(wstatus));
```

**Comment by Profpatsch** [comment]
Pushed the suggested changes, PTAL.

**Pull request merged**